packages/components/src/unit-control/index.native.js

```diff
@@ -105,6 +104,11 @@ function UnitControl( {
 		[ anchorNodeRef?.current ]
 	);
 
+	const getDecimal = ( step ) => {
+		const stepToString = step;
+		return step === 1 ? 0 : stepToString.toString().length - 2;
```

Comment:
I don't follow exactly what this is accomplishing. Is this toggling between zero places and two places? Why was this put in place and `decimalNum` removed?

Reply:
If I understand correctly, it appears to be moved here so that the `decimalNum` value can be calculated on the fly from the `UnitControl`'s `step` value as that may change based on the current unit selection. 

Perhaps a comment in code here could help clarify?

Reply:
I updated the code to be a bit more self explanatory and more readable (how I would do things in my head vs a strange equation). 

Instead of passing in the `decimalNum` via a prop, I thought it was better to use the step to calculate it. This makes things work a bit more as expected and reduces bugs. Since you don't end up in situations where the `step` and the `decimalNum` are off and the user incroments the step but the numbers show are not going up. This happens when the decimalNum is set to 1 ( 2.2 ) and the step is set to 0.01 for example. 

Currently if you don't define the step they are pulled from CSS_UNITS which has different values for each unit so just passing in a single decimalNum you endup with things not working as expected. .

Reply:
@enejb thank you for taking the time to explain. 🙇🏻  The changes make a lot of sense to me now, and I agree they will help avoid introducing bugs. 💯